File: 14_longest_common_prefix/longestCommonPrefix.go
func longestCommonPrefix(strs []string) string {
    sort.Strings(strs)

    var res strings.Builder

    first, last := strs[0], strs[len(strs) - 1]

    n := min(len(first), len(last))

    for i := 0; i < n; i++ {
        if first[i] != last[i] {
            return res.String()
        }

        res.WriteByte(first[i])
    }

    return res.String()
}

File: 14_longest_common_prefix/longestCommonPrefix.ts
function longestCommonPrefix(strs: string[]): string {
  strs.sort();

  let res = "";

  const first = strs.at(0) || "";
  const last = strs.at(-1) || "";

  const n = Math.min(first.length, last.length);

  for (let i = 0; i < n; i++) {
    if (first[i] != last[i]) return res;

    res += first[i];
  }

  return res;
}
